test/k8sT/Policies.go

```diff
@@ -281,5 +281,240 @@ var _ = Describe("K8sPolicyTest", func() {
+		Expect(pods).Should(BeTrue())
+		Expect(err).Should(BeNil())
+
+		By("Getting information about pods in qa namespace")
+		res = kubectl.Exec("kubectl get pods -n qa -o wide")
+		log.Infof("%s", res.GetStdOut())
+
+		By("Getting information about pods in development namespace")
+		res = kubectl.Exec("kubectl get pods -n development -o wide")
+		log.Infof("%s", res.GetStdOut())
+
+		By("Getting information about backend service in development namespace")
+		res = kubectl.Exec("kubectl describe svc -n development backend")
+		log.Infof("%s", res.GetStdOut())
+
+		frontendPod, err := kubectl.GetPods(qaNs, formatLabelArgument("id", "client")).Filter(podNameFilter)
+		Expect(err).Should(BeNil())
+
+		By(fmt.Sprintf("%s", frontendPod))
+
+		backendPod, err := kubectl.GetPods(developmentNs, formatLabelArgument("id", "server")).Filter(podNameFilter)
+		Expect(err).Should(BeNil())
+
+		By(fmt.Sprintf("%s", backendPod))
+
+		backendSvcIP, err := kubectl.Exec("kubectl get svc -n development -o json").Filter("{.items[*].spec.clusterIP}")
+		Expect(err).Should(BeNil())
+
+		By(fmt.Sprintf("Backend Service IP: %s", backendSvcIP.String))
+
+		By("Running tests WITHOUT Policy / Proxy loaded")
+
+		By(fmt.Sprintf("running curl %s:80 from pod %s (should work)", backendSvcIP, frontendPod))
+		returnCode := kubectl.Exec(fmt.Sprintf("kubectl exec -n qa -i %s -- curl -s -o /dev/null -w \"%%{http_code}\" http://%s:80/", frontendPod, backendSvcIP)).GetStdOut()
+
+		Expect(returnCode).Should(Equal("200"), "Unable to connect between %s and %s:80/", frontendPod, backendSvcIP)
+
+		l7StressTest := func() {
+			By("L7 Stresstest")
+			By("Loading Policies into Cilium")
+
+			policyPath := kubectl.ManifestGet("cnp-l7-stresstest.yaml")
+			policyCmd := "cilium policy get io.cilium.k8s-policy-name=l7-stresstest"
+
+			_, err = kubectl.CiliumPolicyAction(helpers.KubeSystemNamespace, policyPath, helpers.KubectlCreate, helpers.HelperTimeout)
+			Expect(err).Should(BeNil(), fmt.Sprintf("Error creating resource %s: %s", policyPath, err))
+
+			defer func() {
+				_, err := kubectl.CiliumPolicyAction(helpers.KubeSystemNamespace, policyPath, helpers.KubectlDelete, helpers.HelperTimeout)
+				Expect(err).Should(BeNil(), fmt.Sprintf("Error deleting resource %s: %s", policyPath, err))
+
+				By("Checking that all policies were deleted in Cilium")
+				output, err := kubectl.ExecPodCmd(helpers.KubeSystemNamespace, ciliumPodK8s1, policyCmd)
+				Expect(err).Should(Not(BeNil()), "policies should be deleted from Cilium: policies found: %s", output)
+			}()
+
+			output, err := kubectl.ExecPodCmd(helpers.KubeSystemNamespace, ciliumPodK8s1, "cilium policy get")
+			Expect(err).Should(BeNil(), fmt.Sprintf("output of \"cilium policy get\": %s", output))
+
+			By("Running tests WITH Policy / Proxy loaded")
+
+			testConnectivity(backendSvcIP.String(), frontendPod.String())
+		}
+
+		l7StressTest()
+
+		crossNamespaceTest := func() {
```

Comment:
Unnecessary function as well?

Reply:
Not unnecessary; the defer call within it ensures that all policies are deleted within Cilium at the end of the function.